docs/configuration/unit_testing_rules.md

```diff
@@ -275,3 +289,49 @@ groups:
         summary: "Instance {{ $labels.instance }} down"
         description: "{{ $labels.instance }} of job {{ $labels.job }} has been down for more than 5 minutes."
 ```
+
+### Time within tests
+
+It should be noted that in all tests, either in `alert_test_case` or
+`promql_test_case`, the output from all functions related to the current time,
+for example the `time()` and `day_of_*()` functions, will output a consistent value
+for tests.
+
+By default, at the start of the test evaluation, `time()` returns 0 (Unix epoch:
+January 1, 1970 00:00:00 UTC). The `eval_time` field specifies a duration relative
+to `start_timestamp`, so by default `time()` will return a value of `0 + eval_time`.
+
+#### Configuring the start timestamp
+
+You can configure a custom start timestamp for your tests by setting the `start_timestamp`
+field in your test group. This field accepts either:
+- A Unix timestamp (e.g., `1609459200` for January 1, 2021 00:00:00 UTC)
+- An RFC3339 formatted timestamp (e.g., `"2021-01-01T00:00:00Z"`)
+
+When you set `start_timestamp`:
+- All `input_series` samples will be timestamped starting from `start_timestamp`
+- The `eval_time` field in test cases is interpreted as a duration relative to `start_timestamp`
+- The `time()` function will return `start_timestamp + eval_time`
+
+#### Testing alerts with time functions
+
+If you need to write tests for alerts that use functions relating to the current
+time, make sure that the values given to your `input_series` are placed far
+enough in the past, relative to the evaluation time described above. The values
+can for example be negative timestamps so that with a very small `eval_time` the
+alert can be expected to trigger.
+
+Another method that's known to work is to instead bump `eval_time` in the future
+so that the timestamp output by `time()` will be a higher value and the values
+in `input_series` will be far enough apart from that point in time so that the
+alerts will trigger. This method has the downside of making promtool generate a
+timeseries database that contains a value for each `input_series` for each
+`interval` for the given test. This can become very slow relatively easily and
+can end up consuming a lot of RAM for running your test. By instead using values
+for `input_series` relative to the timestamp described above even though the
+values go into negative numbers, you can keep `eval_time` fairly lower and avoid
+making your tests run very slowly.
+
+Alternatively, you can use the `start_timestamp` field to set a specific base time
```

Comment:
I think this needs to be moved up to first place as *the* solution and have an example. Maybe say that

"
If you need to write a test that includes time series having timestamps as the value and you want to use the `time()` or `day_of_*()` functions, then use the `start_timestamp` field to set at what time the loading and evaluation of sample should start. For example:

...

This method avoids having to use negative timestamp or loading huge amounts of samples to simulate a large current time as returned by `time()`.
"